curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md

```diff
@@ -17,439 +17,530 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
+The first element only.
 
 ---
 
-Placeholder distractor 3
+An empty array.
 
 #### --answer--
 
-Placeholder answer
+A copy of the original array.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What technique can improve recursive performance?
 
 #### --distractors--
 
-Placeholder distractor 1
+Using more base cases.
 
 ---
 
-Placeholder distractor 2
+Adding more parameters.
 
 ---
 
-Placeholder distractor 3
+Using global variables.
 
 #### --answer--
 
-Placeholder answer
+Memoization
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will this recursive function do?
+
+```javascript
+function repeat(str) {
+    return str + repeat(str);
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Return the string twice.
 
 ---
 
-Placeholder distractor 2
+Return undefined.
 
 ---
 
-Placeholder distractor 3
+Create an empty string.
 
 #### --answer--
 
-Placeholder answer
+Cause a stack overflow.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What HTML structure is typically traversed recursively?
 
 #### --distractors--
 
-Placeholder distractor 1
+Style sheets
 
 ---
 
-Placeholder distractor 2
+Meta tags
 
 ---
 
-Placeholder distractor 3
+Script tags
 
 #### --answer--
 
-Placeholder answer
+The DOM tree
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In JavaScript, what keyword can help prevent infinite recursion?
 
 #### --distractors--
 
-Placeholder distractor 1
+`loop`
 
 ---
 
-Placeholder distractor 2
+`console`
 
 ---
 
-Placeholder distractor 3
+`try`
 
 #### --answer--
 
-Placeholder answer
+`if`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does this recursive function do?
+
+```javascript
+function reverseString(str) {
+  if (str === "") return "";
+  return reverseString(str.slice(1)) + str[0];
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Duplicates a string
 
 ---
 
-Placeholder distractor 2
+Removes vowels
 
 ---
 
-Placeholder distractor 3
+Removes spaces
 
 #### --answer--
 
-Placeholder answer
+Reverses a string.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the following code output?
+
+```javascript
+function isEven(n) {
+  if (n === 0) return true;
+  if (n === 1) return false;
+  return isEven(n - 2);
+}
+console.log(isEven(4));
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+an error
 
 ---
 
-Placeholder distractor 2
+undefined
 
 ---
 
-Placeholder distractor 3
+false
 
 #### --answer--
 
-Placeholder answer
+true
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In the following code, what role does the `return` play?
+
+```javascript
+function addUp(n) {
+  if (n === 1) return 1;
+  return n + addUp(n - 1);
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Breaks the loop.
 
 ---
 
-Placeholder distractor 2
+Declares a variable.
 
 ---
 
-Placeholder distractor 3
+Reverses the string.
 
 #### --answer--
 
-Placeholder answer
+Accumulates values
```

Comment:
I am not sure this is the correct response. `return` terminates a function executing and establish the output